src/app/shared/components/protected-component/protected.component.ts

```diff
@@ -0,0 +1,25 @@
+import { Component, Input } from '@angular/core';
+import { ProfileService } from '../../../core/profile/profile.service';
+
+@Component({
+    selector: 'protected',
+    template: require('./protected.template.html'),
+})
+export class ProtectedComponent {
+    @Input() isAuthorized: boolean;
+    @Input() roles: Array<string>;
+
+    constructor(private profileService: ProfileService) {}
+
+    get allowed(): boolean {
+        if (this.isAuthorized) {
+            if (this.profileService.isLoggedIn()) {
```

Comment:
If this code will be changed according to above comment, please don't forget:
1) Rework ```hasUserRole``` method by changing ```return true``` to ```return this.isLoggedIn()```;
2) Rework method ```getUserRole``` to check for user object existence, before trying to get the role.